google/cloud/spanner/internal/transaction_impl.h

```diff
@@ -32,11 +32,16 @@ namespace cloud {
 namespace spanner_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
+struct TransactionContext {
+  std::string const& tag;
```

Comment:
This gives me pause.  Should this be an `absl::string_view`?  Or maybe `std::string` and replace the `tag_` and `seqno_` member variables in `TransactionImpl` with a `TransactionContext context_`?

Reply:
Using a single, member `TransactionContext` wouldn't work.  The sequence number is per-visit, and multiple visitations can be happening concurrently.

We could use `string_view` for the tag, but unless callbacks need to dissect it (which I'll claim they don't, as it is structure-less ... they simply add it to the request), it doesn't really buy us anything over a `const&`.